homeassistant/components/launch_library/sensor.py

```diff
@@ -72,25 +76,55 @@ async def async_setup_entry(
 
     async_add_entities(
         [
-            NextLaunchSensor(coordinator, entry.entry_id, name),
+            NextLaunchSensor(
+                coordinator,
+                entry.entry_id,
+                SensorEntityDescription(
+                    key="next_launch",
+                    icon="mdi:rocket-launch",
+                    name=name,
+                ),
+            ),
         ]
     )
 
 
-class NextLaunchSensor(CoordinatorEntity, SensorEntity):
-    """Representation of the next launch sensor."""
+class LaunchLibraryEntity(CoordinatorEntity, SensorEntity):
+    """Base entity class for launch library sensors."""
 
     _attr_attribution = ATTRIBUTION
-    _attr_icon = "mdi:rocket-launch"
     _next_launch: Launch | None = None
 
     def __init__(
-        self, coordinator: DataUpdateCoordinator, entry_id: str, name: str
+        self,
+        coordinator: DataUpdateCoordinator,
+        entry_id: str,
+        description: SensorEntityDescription,
     ) -> None:
-        """Initialize a Launch Library entity."""
+        """Initialize a Launch Library base entity."""
         super().__init__(coordinator)
-        self._attr_name = name
-        self._attr_unique_id = f"{entry_id}_next_launch"
+        self._attr_name = description.name
```

Comment:
Keep name as it was as an argument.

Reply:
Name is a key in `SensorEntityDescription`?

From homeassistant.helper.entity
```python
@dataclass
class EntityDescription:
    """A class that describes Home Assistant entities."""

    # This is the key identifier for this entity
    key: str

    device_class: str | None = None
    entity_category: EntityCategory | Literal[
        "config", "diagnostic", "system"
    ] | None = None
    entity_registry_enabled_default: bool = True
    force_update: bool = False
    icon: str | None = None
    name: str | None = None
    unit_of_measurement: str | None = None
```

Or am I missunderstanding something?

Reply:
Ignore my post above, i see what you mean now! 😆